students/mattmaeda/mailroom_app/mailroom/test/test_cli.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+"""
+Unit tests for mailroom cli
+
+pytest --cov=mailroom
+"""
+
+from unittest import mock
+from io import StringIO
+import pytest
+from mailroom import cli
+
+THANK_YOU_OUTPUT = """
+Dear Matt,
+
+Thank you for your donation of $100.00.
+
+"""
+
+BAD_THANK_YOU_OUTPUT = """Invalid amount 'x' entered.
+
+Dear Matt,
+
+Thank you for your donation of $100.00.
+
+"""
+
+BAD_THANK_YOU_OUTPUT2 = """Invalid amount '0.001' entered.
+
+Dear Matt,
+
+Thank you for your donation of $100.00.
+
+"""
+
+TEST_REPORT_OUTPUT = """
+Donor Name                                        | Total Given | Num Gifts | Average Gift
+------------------------------------------------------------------------------------------
+Matt                                               $      100.00           1        100.00
+John Adams                                         $        3.00           1          3.00
+Thomas Jefferson                                   $        3.00           1          3.00
+John Quincy Adams                                  $        2.00           1          2.00
+James Madison                                      $        2.00           1          2.00
+George Washington                                  $        1.00           1          1.00
+
+
+
+"""
+
+
+def test_get_selection():
+    """ test get selection """
+    user_input = ["1"]
+    with mock.patch("builtins.input", side_effect=user_input):
+        option = cli.get_selection()
+        assert option == "1"
+
+
+@mock.patch('sys.stdout', new_callable=StringIO)
+def test_send_thank_you(mock_stdout):
+    """ test send thank you """
+    user_input = ["Matt", "100"]
+    with mock.patch("builtins.input", side_effect=user_input):
+        cli.send_thank_you()
+        assert mock_stdout.getvalue() == THANK_YOU_OUTPUT
```

Comment:
Looks like you got sys.stdout patched! I think there were others having difficulties with this, so you might want to post a link to your tests in slack so others can see what you did.